### PR TITLE
Fix --with-fortran=... logic

### LINKED ISSUE #211: Abort if --enable-mpi-fortran=usempif08 is specified but we can't build F08 bindings
As noted by Paul Hargrove in http://www.open-mpi.org/community/lists/devel/2014/07/15347.php, if you

  --enable-mpi-fortran=usempif08

and configure determines that we can't build the F08 bindings, it doesn't abort.

r32354 was an attempt to fix this, but it wasn't quite right.

### PR DESCRIPTION
**Pull request by ggouaillardet**
This commit fixes #211 
@jsquyres could you please review it before it gets merged into the trunk ?

### CODE REVIEW

File: opal/mca/hwloc/hwloc191/hwloc/config/hwloc.m4 [hunk 1]
```diff
@@ -1306,8 +1308,8 @@ AC_DEFUN([_HWLOC_CHECK_DECL], [
   AC_REQUIRE([AC_PROG_CC])
   AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
        [AC_INCLUDES_DEFAULT([$4])
-       $1(int,long,int,long,int,long,int,long,int,long);],
-       [$1(1,2,3,4,5,6,7,8,9,10);])],
+       void * $1;],
+       )],
```
Comment (jsquyres) [inline opal/mca/hwloc/hwloc191/hwloc/config/hwloc.m4]: I think this hwloc.m4 change is not supposed to be part of this patch, right?

**Comment by jsquyres** [comment]
@ggouaillardet I'm truly sorry it has taken me *months* to look at this.  :-(

Overall, this is great stuff.  Thank you!

I had two minor nits:

1. What looks like an accidental inclusion of the hwloc.m4 stuff (noted above).
1. Some other minor nits that it was easier just to file a pull request against your tree than to described here -- see https://github.com/ggouaillardet/ompi/pull/1

**Comment by jsquyres** [comment]
@ggouaillardet One more thing: you might want to edit your commit message to say "Fixes #211" so that it automatically closes the ticket when this PR is merged.

**Comment by jsquyres** [comment]
@ggouaillardet Any progress on this PR, perchance?

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-master-pr/504/

**Build Log**
*last 50 lines*

 ```
GitHub pull request #228 of commit 86650eadc9d5a62c9e2c61e4c87eb9bfde5119ce.
Setting status of 86650eadc9d5a62c9e2c61e4c87eb9bfde5119ce to PENDING with url http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr/504/ and message: Build started.
[EnvInject] - Loading node environment variables.
Building on master in workspace /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace
Wiping out workspace first.
Cloning the remote Git repository
Cloning repository https://github.com/open-mpi/ompi
 > /usr/bin/git init /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace # timeout=60
Fetching upstream changes from https://github.com/open-mpi/ompi
 > /usr/bin/git --version # timeout=60
using .gitcredentials to set credentials
 > /usr/bin/git config --local credential.helper store --file=/tmp/git2522661632350788868.credentials # timeout=60
 > /usr/bin/git -c core.askpass=true fetch --tags --progress https://github.com/open-mpi/ompi +refs/heads/*:refs/remotes/origin/* # timeout=60
 > /usr/bin/git config --local --remove-section credential # timeout=60
 > /usr/bin/git config remote.origin.url https://github.com/open-mpi/ompi # timeout=60
 > /usr/bin/git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=60
 > /usr/bin/git config remote.origin.url https://github.com/open-mpi/ompi # timeout=60
Pruning obsolete local branches
Fetching upstream changes from https://github.com/open-mpi/ompi
using .gitcredentials to set credentials
 > /usr/bin/git config --local credential.helper store --file=/tmp/git3216641375903993041.credentials # timeout=60
 > /usr/bin/git -c core.askpass=true fetch --tags --progress https://github.com/open-mpi/ompi +refs/pull/*:refs/remotes/origin/pr/* --prune # timeout=60
 > /usr/bin/git config --local --remove-section credential # timeout=60
 > /usr/bin/git rev-parse 86650eadc9d5a62c9e2c61e4c87eb9bfde5119ce^{commit} # timeout=60
Merging Revision 86650eadc9d5a62c9e2c61e4c87eb9bfde5119ce (detached) to origin/master, UserMergeOptions{mergeRemote='origin', mergeTarget='master', mergeStrategy='default', fastForwardMode='--ff'}
 > /usr/bin/git rev-parse origin/master^{commit} # timeout=60
 > /usr/bin/git config core.sparsecheckout # timeout=60
 > /usr/bin/git checkout -f origin/master # timeout=20
 > /usr/bin/git merge --ff 86650eadc9d5a62c9e2c61e4c87eb9bfde5119ce # timeout=60
 > /usr/bin/git config core.sparsecheckout # timeout=60
 > /usr/bin/git checkout -f 86650eadc9d5a62c9e2c61e4c87eb9bfde5119ce # timeout=20
ERROR: Branch not suitable for integration as it does not merge cleanly: Could not merge AnyObjectId[86650eadc9d5a62c9e2c61e4c87eb9bfde5119ce]
TAP Reports Processing: START
Looking for TAP results report in workspace using pattern: **/*.tap
Did not find any matching files.
Anchor chain: could not read file with links: /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/jenkins_sidelinks.txt (No such file or directory)
[copy-to-slave] The build is taking place on the master node, no copy back to the master will take place.
Setting commit status on GitHub for https://github.com/open-mpi/ompi/commit/86650eadc9d5a62c9e2c61e4c87eb9bfde5119ce
[BFA] Scanning build for known causes...
[BFA] No failure causes found
[BFA] Done. 0s
Setting status of 86650eadc9d5a62c9e2c61e4c87eb9bfde5119ce to FAILURE with url http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr/504/ and message: Build finished.
```

Test FAILed.

**Comment by ggouaillardet** [comment]
@jsquyres i am working on it (since the PR is quite old, git rebase ended up with many conflicts ...)

**Comment by ggouaillardet** [comment]
@jsquyres this PR should be ready for prime time now

**Comment by mellanox-github** [comment]
* Coverity found 903 errors for all: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/all_505/output/errors/index.html

* Coverity found 5 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/oshmem_505/output/errors/index.html

* Coverity found 2 errors for fca: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/fca_505/output/errors/index.html <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-master-pr/505/
Test PASSed.

File: opal/mca/hwloc/hwloc191/hwloc/config/hwloc.m4 (hunk 1 above)
Comment (jsquyres) [inline opal/mca/hwloc/hwloc191/hwloc/config/hwloc.m4]: @ggouaillardet Can you back this out / squash the removal into the commit that added it?

**Comment by jsquyres** [comment]
Due to inlining, I think it might be hard to find the comment I just made: can you remove the hwloc change that accidentally crept into this PR?

Then I think it'll probably be good to do (sorry for the massive delay with this!).

**Comment by mellanox-github** [comment]
* Coverity found 902 errors for all: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/all_509/output/errors/index.html

* Coverity found 5 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/oshmem_509/output/errors/index.html

* Coverity found 2 errors for fca: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/fca_509/output/errors/index.html <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-master-pr/509/
Test PASSed.

**Comment by mellanox-github** [comment]
* Coverity found 911 errors for all: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/all_511/output/errors/index.html

* Coverity found 5 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/oshmem_511/output/errors/index.html

* Coverity found 2 errors for fca: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/fca_511/output/errors/index.html <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-master-pr/511/
Test PASSed.

**Pull request merged**